packages/@uppy/xhr-upload/src/index.ts

```diff
@@ -34,78 +98,73 @@ function buildResponseError (xhr, err) {
  * Set `data.type` in the blob to `file.meta.type`,
  * because we might have detected a more accurate file type in Uppy
  * https://stackoverflow.com/a/50875615
- *
- * @param {object} file File object with `data`, `size` and `meta` properties
- * @returns {object} blob updated with the new `type` set from `file.meta.type`
  */
-function setTypeInBlob (file) {
+function setTypeInBlob<M extends Meta, B extends Body>(file: UppyFile<M, B>) {
   const dataWithUpdatedType = file.data.slice(0, file.data.size, file.meta.type)
   return dataWithUpdatedType
 }
 
-export default class XHRUpload extends BasePlugin {
+const defaultOptions = {
+  endpoint: '',
+  formData: true,
+  fieldName: 'file',
+  method: 'post',
+  allowedMetaFields: null,
+  responseUrlFieldName: 'url',
+  bundle: false,
+  headers: {},
+  timeout: 30 * 1000,
+  limit: 5,
+  withCredentials: false,
+  responseType: '',
+  getResponseData(responseText) {
+    return JSON.parse(responseText)
+  },
+  getResponseError(_, response) {
+    let error = new Error('Upload error')
+
+    if (isNetworkError(response)) {
+      error = new NetworkError(error, response)
+    }
+
+    return error
+  },
+  validateStatus(status) {
+    return status >= 200 && status < 300
+  },
+} satisfies XhrUploadOpts<any, any>
+
+type Opts<M extends Meta, B extends Body> = DefinePluginOpts<
+  XhrUploadOpts<M, B>,
+  keyof typeof defaultOptions
+>
+
+interface OptsWithHeaders<M extends Meta, B extends Body> extends Opts<M, B> {
+  headers: Record<string, string>
+}
+
+export default class XHRUpload<
+  M extends Meta,
+  B extends Body,
+> extends BasePlugin<Opts<M, B>, M, B> {
   // eslint-disable-next-line global-require
   static VERSION = packageJson.version
 
-  constructor (uppy, opts) {
-    super(uppy, opts)
-    this.type = 'uploader'
-    this.id = this.opts.id || 'XHRUpload'
-    this.title = 'XHRUpload'
```

Comment:
Just to be sure, we're getting rid of the title property on the none-UIPlugins plugins, correct?

Reply:
Indeed, `title` was just a copy paste mistake on non-UI plugins.